messages.py

```diff
@@ -0,0 +1,53 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta, timezone
+
+import constants
+
+# Messages defined here:
+# Submit                                              : /submit <code> <day> <part>
+# Help:                                               : /help
+# Info:                                               : /info
+# Leaderboard                                         : /best <day> <part>
+
+@dataclass
+class SubmitMessage():
+    msg: object
+    code: str
+    day: int
+    part: int
+
+    def parse(msg):
```

Comment:
FIXME: im unclear why the argument here is called msg
this should be called self if its a method on the type
and if its a staticmethod/classmethod you should use the `@staticmethod` or `@classmethod` decorators, which will always call it as a static or class method regardless if you are calling on the type itself or an instance

it looks like we want `@staticmethod` here, which is equivalent to a structs method with no self argument in rust

Reply:
fixed